src/main/java/com/pubnub/api/models/consumer/history/PNFetchMessageItem.java

```diff
@@ -30,12 +31,13 @@ public class PNFetchMessageItem {
 
     @SerializedName("message_type")
     @Getter(AccessLevel.NONE)
-    private final String messageType;
-    private int getMessageType() {
-        if (messageType == null || messageType.isEmpty()) {
-            return 0;
-        } else {
-            return Integer.parseInt(messageType);
+    private final Integer messageType;
+
+    public HistoryMessageType getMessageType() {
+        try {
```

Comment:
I wrote a test:

```
@Test
public void testFetchSingleChannel_includeMessageTypeIsFalse() throws PubNubException {
    final String expectedChannelName = random();

    publishMixed(pubNub, 10, expectedChannelName);

    final PNFetchMessagesResult fetchMessagesResult = pubNub.fetchMessages()
            .channels(Collections.singletonList(expectedChannelName))
            .includeMessageType(false)
            .sync();

    pause(3);

    assert fetchMessagesResult != null;
    for (PNFetchMessageItem messageItem : fetchMessagesResult.getChannels().get(expectedChannelName)) {
        assertEquals(HistoryMessageType.Message ,messageItem.getMessageType());
    }

}
```

The thing is that following assertion passes:
`assertEquals(HistoryMessageType.Message ,messageItem.getMessageType());`
but in test there is `.includeMessageType(false)`
I would expect messageItem.getMessageType() to return null in this case ?


We could add to PNFetchMessageItem class:
```
    @Getter(AccessLevel.PRIVATE)
    private final boolean includeMessageType;
```

and then something like:

```
    public HistoryMessageType getMessageType() {
        return includeMessageType ? HistoryMessageType.of(messageType) : null;
    }
```

What do you think?

Reply:
you're right! good catch. fixed